pkg/graph/resolvers/mto_solution.go

```diff
@@ -110,7 +115,32 @@ func MTOSolutionCreateCommon(
 		return nil, err
 	}
 
-	return storage.MTOSolutionCreate(store, logger, mtoSolution)
+	return sqlutils.WithTransaction[models.MTOSolution](store, func(tx *sqlx.Tx) (*models.MTOSolution, error) {
+		solution, err := storage.MTOSolutionCreate(tx, logger, mtoSolution)
```

Comment:
This is the section of code I'm referring to. You are currently looping through and creating links, and calling a store method for each link.  The solution for linking solutions to a milestone that takes place in `MTOMilestoneUpdateLinkedSolutionsWithTX` is reusable. It also reduces the amount of store calls. 

Currently, this code would not be useable for the update methods. By having a single resolver which can update all the milestones linked to a solution, we could reuse the logic for update. Otherwise, we'd need to refactor this when we implement the functionality for editing a solution.